System/DataModule.cpp

```diff
@@ -1,764 +1,497 @@
+					foundAny = true;
+				}
+			}
+		} else {
+			// Look only in one specific class (which will get all derived classes' entities too!)
+			// Look for the class entry
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(type);
+			if (clsItr == m_TypeMap.end()) { return 0; }
+
+			RTEAssert(!clsItr->second.empty(), "DataModule has class entry without instances in its map!?");
+
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				// Get the grouped entities, without transferring ownership
+				if (instItr->second->IsInGroup(group)) {
+					entityList.push_back(instItr->second);
+					foundAny = true;
+				}
+			}
+		}
+		return foundAny;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::GetAllOfType(std::list<Entity *> &entityList, std::string type) {
+		if (type.empty()) { return false; }
+
+		// Look for the class entry
+		map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(type);
+		if (clsItr == m_TypeMap.end()) { return false; }
+
+		RTEAssert(!clsItr->second.empty(), "DataModule has class entry without instances in its map!?");
+
+		// Found something, so let's add it to the list!
+		for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+			// Get the entities, without transferring ownership
+			entityList.push_back(instItr->second);
+		}
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::AddMaterialMapping(int fromID, int toID) {
+		RTEAssert(fromID > 0 && fromID < c_PaletteEntriesNumber && toID > 0 && toID < c_PaletteEntriesNumber, "Tried to make an out-of-bounds Material mapping");
+
+		bool clear = m_MaterialMappings[fromID] == 0;
+		m_MaterialMappings[fromID] = toID;
+
+		return clear;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int DataModule::LoadScripts() {
+		if (m_ScriptPath != "") { return g_LuaMan.RunScriptFile(m_ScriptPath); }
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void DataModule::ReloadAllScripts() {
+		for (std::list<PresetEntry>::iterator itr = m_PresetList.begin(); itr != m_PresetList.end(); ++itr) { (*itr).m_pEntityPreset->ReloadScripts(); }
+		LoadScripts();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	Entity * DataModule::GetEntityIfExactType(const std::string &exactType, const std::string &instanceName) {
```

Comment:
Goddamnit data!!!
This is exactly the same as `GetEntityPreset`, except that it takes `const std::string &exactType` and `const std::string &instance` instead of `std::string exactType`, `std::string instance`.

I would like one of these (probably this one) deleted and replaced with GetEntityPreset because jesus christ data wtf.

Reply:
Unfortunately there is one slight difference I overlooked. GetEntityPreset returns a const Entity * while this returns a non-const one. Need to investigate a little to see if GetEntityPreset can reasonably return a non-const one (maybe this would be a problem with lua?) and kill this off if it can, or rename this if it can't. I've put in a todo to that effect.

Reply:
I don't know what you're on about here so open an issue for it or leave it be.
¯\\\_(ツ)_/¯

Reply:
Ooops, this is on me. I forgot to do the followup of actually opening an issue for this. See https://github.com/cortex-command-community/Cortex-Command-Community-Project-Source/issues/87